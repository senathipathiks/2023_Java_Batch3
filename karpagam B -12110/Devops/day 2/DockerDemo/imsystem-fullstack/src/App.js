import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Login from './pages/Login';


import './App.css';
import './index.css';
import AddPet from './pages/AddPet';
import AddUser from './pages/AddUser';
import PetHome from './index/PetHome';
import UserHome from './index/UserHome';
import AppointmentHome from './index/AppointmentHome';


function App() {
  return (
    <div className="App">
      <Router>
        <Routes>
          <Route exact path="/" element={<Login />}></Route>

          <Route exact path="/home" element={<PetHome />}></Route>
          <Route exact path="/user" element={<UserHome />}></Route>

          <Route exact path="/app" element={<AppointmentHome/>}></Route>

          <Route exact path="/addpet" element={<AddPet />}></Route>
          <Route exact path="/addbrand" element={<AddUser/>}></Route>






        </Routes>
      </Router>

    </div>
  );
}

export default App;
