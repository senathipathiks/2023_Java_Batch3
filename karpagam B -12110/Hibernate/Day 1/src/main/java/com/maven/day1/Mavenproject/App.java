package com.maven.day1.Mavenproject;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;



/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
       Student s1=new Student(3,"Anu","Sivakasi");
    	
      Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
      SessionFactory factory= cfg.buildSessionFactory();
      Session session=factory.openSession();
      Transaction t=session.beginTransaction();
      session.save(s1);
      t.commit();
    }
}