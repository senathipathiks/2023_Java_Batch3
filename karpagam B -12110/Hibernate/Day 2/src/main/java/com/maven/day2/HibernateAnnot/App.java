package com.maven.day2.HibernateAnnot;


import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

import java.util.List;
public class App 
{
    public static void main( String[] args )
    {
       Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
       SessionFactory factory=cfg.buildSessionFactory();
       Session session =factory.openSession();
       Transaction t=session.beginTransaction();
       
       
      //****insert operation
//       Employee e1=new Employee();
//       e1.setEname("boo");
//       e1.setEsal("thousand");
//       session.save(e1);
       
       
      //****update operation
//        Employee temp=session.find(Employee.class, 1);
//        temp.setEname("kavi");
//        session.update(temp); // or we can use merge instead of update
       
      //****delete operation
//         Employee temp=session.find(Employee.class, 4);
//         session.delete(temp);
//         
      //****Find operation
       
         List <Employee> empList=session.createQuery("from Employee").list();
         for (Employee e :empList) {
        	 System.out.println(e);
			
		}
         System.out.println("Done");

       
       
       
        
        
        t.commit();
        
   
       
       
      
    }
}
