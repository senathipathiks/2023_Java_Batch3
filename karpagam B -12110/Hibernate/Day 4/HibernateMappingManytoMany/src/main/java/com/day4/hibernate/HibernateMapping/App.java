package com.day4.hibernate.HibernateMapping;



import java.util.ArrayList;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory= cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
       Answer a1=new Answer();
       a1.setAnswername("Answer 1");
       a1.setPostedBy("kaviya");
       
       Answer a2=new Answer();
       a2.setAnswername("Answer 2");
       a2.setPostedBy("Sri");
       
       Question q1=new Question();  
       q1.setQname("What is Answer?");  
       ArrayList<Answer> l1=new ArrayList<Answer>();  
       l1.add(a1);  
       l1.add(a2);  
       q1.setAnswers(l1);  
       
       Answer a3=new Answer();
       a3.setAnswername("Answer 3");
       a3.setPostedBy("Anam");
       
       Answer a4=new Answer();
       a4.setAnswername("Answer 4");
       a4.setPostedBy("Vini");
       
       Question q2=new Question();  
       q2.setQname("What is Answer?");  
       ArrayList<Answer> l2=new ArrayList<Answer>();  
       l2.add(a3);  
       l2.add(a4);  
       q2.setAnswers(l2); 

        session.save(q1);
        session.save(q2);
        
        t.commit();
        System.out.println("Done");


        
        

        
            }
}
