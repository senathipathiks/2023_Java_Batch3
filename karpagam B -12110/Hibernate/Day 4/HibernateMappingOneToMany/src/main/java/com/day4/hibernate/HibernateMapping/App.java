package com.day4.hibernate.HibernateMapping;



import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory= cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        Department d1=new Department(123,"sales");
        Department d2=new Department(124,"Finance");
        Employee e1=new Employee(107,"vinitha",d2);
        Employee e2=new Employee(108,"kaviya",d1);
        Employee e3=new Employee(109,"jovika",d1);
        
        session.persist(e1);
        session.persist(e2);
        session.persist(e3);
        t.commit();


        
        

        
            }
}
