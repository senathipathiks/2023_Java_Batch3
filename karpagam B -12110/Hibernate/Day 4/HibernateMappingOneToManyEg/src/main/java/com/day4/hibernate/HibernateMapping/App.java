package com.day4.hibernate.HibernateMapping;

import java.util.ArrayList;
import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;
public class App 
{
	public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory= cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        Answer a1=new Answer();
        a1.setAnswername("It is ORM Framework");
        a1.setPostedBy("kp");
        
        Answer a2=new Answer();
        a2.setAnswername("It is a object relational modal");
        a2.setPostedBy("Edwin");
        
        ArrayList<Answer> l1=new ArrayList<Answer>();    
        l1.add(a1);    
        l1.add(a2);      
        
        Question q1=new Question();    
        q1.setQname("What is Hibernate?");    
        q1.setAnswers(l1);  
        session.save(q1);
        
        t.commit();
        System.out.println("Done");
}
}
