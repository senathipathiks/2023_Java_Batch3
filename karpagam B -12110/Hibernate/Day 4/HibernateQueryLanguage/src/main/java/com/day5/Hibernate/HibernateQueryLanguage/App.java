package com.day5.Hibernate.HibernateQueryLanguage;

import java.util.List;
import java.util.Random;

import javax.persistence.Query;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;


/**
 * Hello world!
 *
 */
public class App 
{ 
	@SuppressWarnings({"unchecked","deprecation"}) //it will not give warning for unused variables
    public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session =factory.openSession();
        Transaction t=session.beginTransaction();
        
        Random r=new Random();//Generate a random number
        
//        for(int i=1;i<30;i++) {
//        	Employee emp=new Employee();
//        	emp.setEid(i);
//        	emp.setEname("Name"+i);
//        	emp.setDept(r.nextInt(4)); // it will generate random number from 0 to 4
//        	}
        
        //to list all the data in the table (HQL)
        
//        
//        Query query=session.createQuery("from Employee"); // we have to mention class name i.e Employee
//        List<Employee> employee=query.getResultList();
//        for(Employee e:employee) {
//        	System.out.println(e);
//
//        }
//        
        
        
         //Get the value which the deptno is greaterthan 1
//        Query query=session.createQuery("from Employee where dept>1");
//        List<Employee> employee=query.getResultList();
//        for(Employee e:employee) {
//      	System.out.println(e);
//      	}
        
        
//        //show the id which the id =20
//        Query query=session.createQuery("from Employee where eid=20 ");
//        List<Employee> employee=query.getResultList();
//    	  System.out.println(employee);
//        
        
       //select Query Execution
//          Query query=session.createQuery("select eid,dept,ename from Employee where eid=4");
//          Object[] employee=(Object[])((org.hibernate.query.Query)query).uniqueResult();
//
//          System.out.println(employee[0]+":"+employee[1]+":"+employee[2]);
        
      // select with where class
//      Query query=session.createQuery("select eid,dept,ename from Employee where dept>1");
//      List<Object[]> employees=(List<Object[]>)query.getResultList();
//      for(Object[] employee :employees) {
//    	  System.out.println(employee[0]+":"+employee[1]+":"+employee[2]);
//      }
//      
        
      //Aggregate function
//      Query query=session.createQuery("select avg(eid) from Employee where dept>0");
//      List employees=(List)query.getResultList();
//      for(Object count:employees) {
//    	  System.out.println(count);
//      }
      
        
        //UPDATE THE ELEMENT
//       Query query=session.createQuery("update Employee set ename=:n where eid=:i");
//       query.setParameter("n", "Anand");
//       query.setParameter("i", 7 );
//       int status=query.executeUpdate();
//       if(status>0) {
//    	   System.out.println("Record Updated Successfully !!");
//       }
//       else {
//    	   System.out.println("Record updation failure !!");
//       }
      
        
        //DELETE THE ELEMENT
        Query query=session.createQuery("delete from Employee where eid=27");
        int status=query.executeUpdate();
        if(status>0) {
        	System.out.println("Record deleted Successfully !!");
        }
        else {
        	System.out.println("Record deletion failure");
        }
        

      

        
        
         t.commit();
}
}
