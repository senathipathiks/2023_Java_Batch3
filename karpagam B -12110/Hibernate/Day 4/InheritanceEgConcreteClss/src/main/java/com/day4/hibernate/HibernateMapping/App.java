package com.day4.hibernate.HibernateMapping;



import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory= cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
//      Employee e1=new Employee(103,"Sri");

        Employee e2=new RegularEmp(101,"Sri",2000.00f,1000);
        Employee e3=new ContractEmp(102,"Jo",500.00f,"1 Years");

        
//      session.save(e1);
      session.save(e2);
      session.save(e3);
      t.commit();
      System.out.println("Done !!");

        
        
        
    }
}
