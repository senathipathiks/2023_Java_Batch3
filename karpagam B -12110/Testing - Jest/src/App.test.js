import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';

//test suite

describe('Test suite for App', () => {

  //this will create snapshot for the app.js component
  // test('Snapshot test', () => {
  //   const comp = renderer.create(<App/>);
  //   const tree = comp.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  test('render hello world', () => {
    render(<App />);
    const linkElement = screen.getByText(/Hello world!!/i);
    //check whether it is available in the web page( learn react)
    expect(linkElement).toBeInTheDocument();
  });


// //either we can use test or it to describe the test cases
  it('renders this is the app created using react', () => {
    render(<App />);
    const linkElement = screen.getByText(/This is the app created using react/i);
    //check whether it is available in the web page( learn react)
    expect(linkElement).toBeInTheDocument();
  });

  it('renders created by kp', () => {
  //   //it will render on virtual dom not on real dom
    render(<App />);
    const linkElement = screen.getByText(/Created by kp !/i);
     //check whether it is available in the web page( learn react)
    expect(linkElement).toBeInTheDocument();
  });

  //failed test case
  //we can use test.skip to skip the test case
  // it('renders created by kp for failed test case', () => {
  //   render(<App />);
  //   const linkElement = screen.getByText("Created by kp");
  //   expect(linkElement).toBeInTheDocument();
  // });

});



