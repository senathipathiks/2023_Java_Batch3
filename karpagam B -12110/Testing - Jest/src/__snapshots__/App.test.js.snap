// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for App Snapshot test 1`] = `
<div
  className="App"
>
  <h1>
    Hello world!!
  </h1>
  <br />
  <h2>
    This is the app created using react
  </h2>
  <br />
  <h2>
    Created by kaviya !!
  </h2>
</div>
`;
